javalib/src/main/scala/java/io/InputStreamReader.scala

```diff
@@ -72,6 +72,7 @@ class InputStreamReader(private[this] var in: InputStream,
     ensureOpen()
 
     if (outBuf.hasRemaining) outBuf.get()
+    else if (in.available() <= 0) -1
```

Comment:
[`ZipInputStreamTest` fails due to this change](https://scala-webapps.epfl.ch/sjsjenkins/blue/organizations/jenkins/scalajs-main/detail/PR-3665/8/pipeline#step-96-log-111).
>[error] Test scala.scalajs.testsuite.javalibex.ZipInputStreamTest.should_read_zip_archives failed: java.lang.AssertionError: expected:<101> but was:<65535>, took 0.010000128 sec
> [error]     at scala.scalajs.testsuite.javalibex.ZipInputStreamTest.expectStrEntry$1(/localhome/jenkins/sjsjenkins/workspace/scalajs-main_PR-3665/javalib-ex-test-suite/target/scala-2.10/file:/localhome/jenkins/sjsjenkins/workspace/scalajs-main_PR-3665/javalib-ex-test-suite/src/test/scala/scala/scalajs/testsuite/javalibex/ZipInputStreamTest.scala:48)

It could be potential issue in `ZipInputStream`.
I will investigate.

Reply:
I think this was due to inappropriate logic of `Reader.skip` and is fixed in https://github.com/scala-js/scala-js/pull/3665/commits/e965d0259653874e150bfebe1bed781850e5c338

Reply:
I think this was due to inappropriate implementation of `Reader#skip` that is inherited by `InputStreamReader`.
Fixed in https://github.com/scala-js/scala-js/pull/3665/commits/e965d0259653874e150bfebe1bed781850e5c338